main/src/ca/uwaterloo/flix/language/phase/Simplifier.scala

```diff
@@ -272,7 +272,7 @@ object Simplifier extends Phase[TypedAst.Root, SimplifiedAst.Root] {
         val e = visitExp(exp)
         val lambdaTyp = Type.mkArrowWithEffect(Type.Unit, eff, e.tpe, loc)
         val lambdaExp = SimplifiedAst.Expression.Lambda(List(), e, lambdaTyp, loc)
-        SimplifiedAst.Expression.Spawn(lambdaExp, lambdaTyp, loc)
+        SimplifiedAst.Expression.Spawn(lambdaExp, tpe, loc)
```

Comment:
Wait. Should it not be `Type.Unit`? Otherwise if you have `spawn 123` then won't it be Int32?

Reply:
the tpe is Unit originating from the typer
```scala
case KindedAst.Expression.Spawn(exp, loc) =>
        val e = visitExp(exp, subst0)
        val tpe = Type.Unit
        val eff = e.eff
        TypedAst.Expression.Spawn(e, tpe, eff, loc)
```
It should probably be a case class constant instead of an argument